music21/stream/tests.py

```diff
@@ -4575,18 +4575,17 @@ def collectAccidentalDisplayStatus(s_inner):
 
         # mStream.show()
 
-        # after transposition all accidentals are reset
-        # note: last d# is not showing in Finale, but this seems to be a
-        # finale error, as the musicxml is the same in all D# cases
+        # after diatonic transposition no accidentals are reset
```

Comment:
I'm still not convinced that we can keep displayStatus after a diatonic transposition:

```
>>> n = note.Note('F')
>>> n.pitch.accidental = pitch.Accidental('natural')
>>> m = stream.Measure([n, note.Note('F#')])
>>> m.makeNotation(inPlace=True)
>>> n.pitch.accidental.displayStatus
False
>>> m2 = m.transpose('A1')
>>> m2[note.Note][0]
<music21.note.Note F#>
>>> print(m2[note.Note][0].pitch.accidental.displayStatus)
None
```

it would be a major mistake to keep that as False after the transposition.

Reply:
Hmmm... This is distressing.  I was imagining that diatonic transpositions were all about changing _key_.  But as you say, the key doesn't necessarily get transposed.  But I really need to be able to transpose to written or sounding pitch (for transposing instruments as well as for ottavas) while maintaining displayStatus, and I believe that should be valid for those cases.  Maybe I need to re-introduce inheritAccidentalDisplay: bool as a parameter, so that those who know they are changing key (or transposing by octaves) can set it True and get the benefit.

Reply:
Are you sure you don't want to be manipulating `displayType` rather than `displayStatus`? Status is quasi-private.

Reply:
What @jacobtylerwalls is exactly what I came to say.  `displayType` determines the rules about when the accidental should be displayed.  'always' (or 'even-tied') is the equivalent of don't hide this accidental ever.  'never' is don't show this accidental ever.  If `displayStatus` is None, then it will recompute it using `displayType` and context.  For `always` and `never` that recomputation is very fast.

Reply:
Latest commit answers this (with `treatAsKeyChange` parameter, set appropriately in `s._transposeByInstrument`).